domain/placement/placement.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package placement
```

Comment:
I think this should go into a `deploy` or `deployment` domain. We'll relocate some other stuff there in time.